examples/convert_ieeg_to_bids.py

```diff
@@ -43,92 +41,63 @@
-column_names = data[0, :]
-info = data[1:, :]
-electrode_tsv = dict()
-for i, name in enumerate(column_names):
-    electrode_tsv[name] = info[:, i].tolist()
-
-# load in channel names
-ch_names = electrode_tsv['name']
-
-# load in the xyz coordinates as a float
-elec = np.empty(shape=(len(ch_names), 3))
-for ind, axis in enumerate(['x', 'y', 'z']):
-    elec[:, ind] = list(map(float, electrode_tsv[axis]))
-
-# %%
-# Now we make a montage stating that the iEEG contacts are in the MNI
-# coordinate system, which corresponds to the `fsaverage` subject in
-# FreeSurfer. For example, one can look at how MNE-Python deals with iEEG data
-# at `Working with SEEG
-# <https://mne.tools/stable/auto_tutorials/misc/plot_seeg.html>`_.
-montage = mne.channels.make_dig_montage(ch_pos=dict(zip(ch_names, elec)),
-                                        coord_frame='mni_tal')
-print(f'Created {len(ch_names)} channel positions')
-print(dict(zip(ch_names, elec)))
-
-# %%
-# We will load a :class:`mne.io.Raw` object and
-# use the montage we created.
-info = mne.create_info(ch_names, 1000., 'ecog')
-raw = mne.io.read_raw_edf(misc_path + '/ecog/sample_ecog.edf')
+raw = mne.io.read_raw_fif(op.join(
+    misc_path, 'seeg', 'sample_seeg_ieeg.fif'))
 raw.info['line_freq'] = 60  # specify power line frequency as required by BIDS
-raw.set_channel_types({ch: 'ecog' for ch in raw.ch_names})
-
-# set the bad channels
-raw.info['bads'].extend(['BTM1', 'BTM2', 'BTM3', 'BTM4', 'BTM5', 'BTM6',
-                         'BTP1', 'BTP2', 'BTP3', 'BTP4', 'BTP5', 'BTP6',
-                         'EKGL', 'EKGR'])
+subjects_dir = op.join(misc_path, 'seeg')  # Freesurfer recon-all directory
 
-# set montage
-# (note that this only works for some channel types: EEG/sEEG/ECoG/DBS/fNIRS)
-raw.set_montage(montage, on_missing='warn')
+# %%
+# Now we a montage with an `ACPC-aligned coordinate system
+# <https://surfer.nmr.mgh.harvard.edu/fswiki/CoordinateSystems>`_.
+# We can either do this in the subject's anatomical space (from their T1
+# image) or we can transform to a template space such as ``fsaverage``
+# that is also an ACPC space. Automated alignment to ACPC has not been
+# implemented in MNE yet, so for saving in individual anatomical space,
+# the T1 has to be aligned to ACPC before finding the coordinates.
+# Then the ACPC-aligned-T1 can be stored using
+# :func:`mne_bids.write_anat`. Then, we can save the ``raw`` object in BIDS.
+
+# estimate the transformation from "head" to "mri" space
+trans = mne.coreg.estimate_head_mri_t('sample_seeg', subjects_dir)
 
 # %%
-# Let us confirm what our channel coordinates look like.
+# Now let's convert the montage to "mri"
+montage = raw.get_montage()
+montage.apply_trans(trans)  # head->mri
 
-# make a plot of the sensors in 2D plane
-raw.plot_sensors(ch_type='ecog')
+# careful, this should only be done in this specific circumstance,
+# it prevents MNE from transforming the coordinate system to "head"
+montage.remove_fiducials()
```

Comment:
I feel like the idea of having to remove fiducials just so one can obtain coordinates in "not-head" system seems weird to me? Wouldn't having more data points always be better?

I wasn't following other issues. Is it possible to make MNE not use "head" coordinate system?

Reply:
The way I understand it the fiducials are just what is used for coordinate transformation calculations so they're not like extra points in that regard. And then everything is standardized to head for things like plotting consistency but unless you want to pass the montage in addition, you're stuck with head. Passing the montage too might not be the worst solution.